packages/qwik-city/middleware/request-handler/polyfill.unit.ts

```diff
@@ -69,6 +69,25 @@ describe('_TextEncoderStream_polyfill tests', () => {
     expect(polyWriter.write({} as any)).toEqual(nativeWriter.write({} as any));
   });
 
+  it("can encode emoji characters just the same as Node.js' implementation", async () => {
+    const polyfillStream = new _TextEncoderStream_polyfill();
+    const nativeStream = new TextEncoderStream();
+
+    const input = '🦊test emoji encoding📦';
+
+    const polyReader = polyfillStream.readable.getReader();
+    const nativeReader = nativeStream.readable.getReader();
+
+    polyfillStream.writable.getWriter().write(input);
+    nativeStream.writable.getWriter().write(input);
```

Comment:
By the way, I found out that `.write` method returns a Promise, so it should be awaited in every other tests, as well as in this one. The tests are passing at their current state and are correct (except for one you testes sequential writing - I don't understand why are you comparing two promises returned by `.write method`), but I will send a PR with the improvement once I figure out why `.write` method freezes tests once when I await it. I'm not much familiar with WritableStream, so not sure what is wrong here.

Reply:
yeah I'm not sure either. the tests will retry x amount of times for anything async before giving up and timing out. do you want to make that test fix into another pr?

Reply:
Yeah, this is just something to improve. The tests are still valid, as far as I can tell (they pass the expectations).

Reply:
ok lets ship this pr then

Reply:
well it looks like the e2e tests are hanging maybe because we need this in?

Reply:
I don't see e2e hanging, but GitHub awaits for status report.

Am I looking to the right place?

Here https://github.com/QwikDev/qwik/actions/runs/9421547564/job/25958622059?pr=6466

and here https://github.com/QwikDev/qwik/actions/runs/9421547564/job/25958622214?pr=6466

Reply:
<img width="924" alt="Screenshot 2024-06-07 at 1 23 09 PM" src="https://github.com/QwikDev/qwik/assets/1016365/657eda2c-96b3-4cdd-b869-5380f4d8479d">

yeah, I'm not sure why it's not merging

Reply:
Maybe the problem on GitHub actions side, this happens sometimes. I suggest we just wait, because this PR is not going anywhere :)

Reply:
fixed it 😊

Reply:
Thank you!